proposals/0016-simd-program-rebatable-account-write-fees.md

```diff
@@ -0,0 +1,637 @@
+adversary with malicious intent can block its competitors by writing or
+read-locking their accounts through a transaction. This attack involves
+carrying out intensive calculations that consume a large number of
+computational units, thereby blocking competitors from performing MEV during
+that particular block, and giving the attacker an unfair advantage. We have
+identified specific transactions that perpetrate this attack and waste
+valuable block space. The malicious transaction write locks multiple token
+accounts and consumes 11.7 million CU i.e around 1/4 the block space. As a
+result, such attacks can prevent users from using their token accounts,
+vote accounts, or stake accounts, and dapps from utilizing the required
+accounts. With the proposed solution, every program, such as the token
+program, stake program, and vote program, can include instructions to employ
+the PRAW fees feature on their accounts and rebate the fees if the user
+initiates the transaction. The attacker will find this option unfeasible as
+they will consume their SOL tokens more rapidly to maintain the attack.
+
+## Security Considerations
+
+If the PRAW fees fee for an account is set too high then we cannot ever mutate
+that account anymore. Maximum fees should be set to around 4.398046511104 *
+10^12 or 2^42 microlamports per CU, assuming that instruction that updates the
+PRAW fees on the accounts will use around 1000 CUs, program owner will need
+roughly 4.3980 SOLs to reupdate the PRAW fees. Optimizing the instruction to
+update the PRAW fees is a must for dapp developers.
+
+For an account that has collected PRAW fees, to transfer these fees
+collected to another account we have to pay PRAW fees to write lock the
+account, we can include a rebate in the transaction. In case of
+any bugs, while transferring PRAW fees from the account to the
+authority, there can be an endless loop where the authority creates a
+transaction to recover collected PRAW fees, with an instruction to pay
+PRAW fees to modify the account and an instruction to rebate. If the
+transaction fails because of the bug, the user fails to recover collected
+fees, in turn increasing PRAW fees collected on the account.
+
+## Backwards Compatibility
+
+This feature does not introduce any breaking changes. The transaction without
+using this feature should work as it is. To use this feature supermajority of
+the validators should move to a branch that implements this feature. Validators
+that do not implement this feature cannot replay the blocks with transactions
+using PRAW fees they also could not validate the block including transactions
+with PRAW fees.
+
+
+## Additional Notes
+
+If the dapps want to rebate PRAW fees they have to implement very
+carefully the logic of rebate. They should be very meticulous before calling
+rebate so that a malicious user could not use this feature to bypass
+PRAW fees. Dapp developers also have to implement additional
```

Comment:
good that you recognized how brittle this proposal is in practice. we need to fix that.  writelock spamming does not come about by demonstration of intelligence by user nor dapp developer.

Reply:
This is not the place to discuss individual's intelligence.

Reply:
ah. _this_ one. apologies this was meant to imply that we're going to get into this situation by greed, not careful thought. i can see how it might be taken as ad hominen if read literally :sweat_smile: